src/main/java/io/kestra/plugin/ldapManager/LdapConnection.java

```diff
@@ -0,0 +1,60 @@
+package io.kestra.plugin.ldapManager;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.Task;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+import jakarta.validation.constraints.NotNull;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+@SuperBuilder
+@Getter
+@NoArgsConstructor
+public abstract class LdapConnection extends Task {
+    @Schema(
+        title = "Hostname",
+        description = "Hostname for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected String hostname;
+
+    @Schema(
+        title = "Port",
+        description = "Port for connection."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    protected Integer port;
```

Comment:
Dynamic properties are always of type String because they can be rendered. You need to render the property and then use something like: `Integer#parseInt`